docs/core/encoding.md

```diff
@@ -69,9 +69,139 @@ typically used for when the data needs to be streamed or grouped together
 
 ### Gogoproto
 
-Modules are encouraged to utilize Protobuf encoding for their respective types.
+Modules are encouraged to utilize Protobuf encoding for their respective types. In the SDK, we use a specific implementation of the Protobuf spec, called [Gogoproto](https://github.com/gogo/protobuf), which offers speed and DX improvements compared to the official [Google implementation](https://github.com/protocolbuffers/protobuf).
 
-#### FAQ
+### Guidelines for protobuf message definitions
+
+In addition to [following official guidelines](https://developers.google.com/protocol-buffers/docs/proto3#simple), we recommend to use these annotations in .proto files when dealing with interfaces:
+
+- fields which accept interfaces should be annotated with `cosmos_proto.accepts_interface`
+  using the same full-qualified name passed as `protoName` to `InterfaceRegistry.RegisterInterface`
+- interface implementations should be annotated with `cosmos_proto.implements_interface`
+  using the same full-qualified name passed as `protoName` to `InterfaceRegistry.RegisterInterface`
+
+### Transaction Encoding
+
+Another important use of Protobuf is the encoding and decoding of
+[transactions](./transactions.md). Transactions are defined by the application or
+the SDK, but passed to the underlying consensus engine in order to be relayed to
+other peers. Since the underlying consensus engine is agnostic to the application,
+it only accepts transactions in the form of raw bytes. The encoding is done by an
+object called `TxEncoder` and the decoding by an object called `TxDecoder`.
+
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc4/types/tx_msg.go#L83-L87
+
+A standard implementation of both these objects can be found in the [`auth` module](../../x/auth/spec/README.md):
+
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc4/x/auth/tx/decoder.go
+
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc4/x/auth/tx/encoder.go
+
+Please also refer to [ADR-020](../architecture/adr-020-protobuf-transaction-encoding.md) on the details of how a transaction is encoded.
+
+### Interface Encoding and Usage of `Any`s
+
+The Protobuf DSL is strongly typed, which can make inserting variable-typed fields difficult. Imagine we want to create a `Profile` protobuf message, which serves as a wrapper over [an account](../basics/accounts.md):
```

Comment:
```suggestion
### Interface Encoding and Usage of `Any`

The Protobuf DSL is strongly typed, which can make inserting variable-typed fields difficult. Imagine we want to create a `Profile` protobuf message that serves as a wrapper over [an account](../basics/accounts.md):
```

Reply:
we can usually use singular, so if we have tick marks around the subject we can omit the s for plural